ecl/eclcc/eclcc.cpp

```diff
@@ -1697,7 +1699,48 @@ void EclCC::generateOutput(EclCompileInstance & instance)
                 addManifestResourcesToArchive(instance.archive, resourceManifestFiles.item(i));
 
             if (optArchive)
+            {
+                if (optCheckDirty)
+                {
+                    Owned<IPipeProcess> pipe = createPipeProcess();
+                    if (!pipe->run("git", "git describe --tags --dirty --long", ".", false, true, false, 0, false))
+                    {
+                        WARNLOG("Failed to run git describe");
+                    }
+                    else
+                    {
+                        try
+                        {
+                            unsigned retcode = pipe->wait();
+                            StringBuffer buf;
+                            Owned<ISimpleReadStream> pipeReader = pipe->getOutputStream();
+                            const size32_t chunkSize = 128;
```

Comment:
Yes, I know - I did look to see if there was somewhere that matched I could call but existing ones were slightly different.